pkg/stringid/stringid.go

```diff
@@ -12,12 +12,15 @@ import (
 	"regexp"
 	"strconv"
 	"strings"
+	"sync"
+	"syscall"
 	"time"
 )
 
 const shortLen = 12
 
 var (
+	randLock     sync.Mutex // No math/rand method should ever be called concurrently
```

Comment:
This lock is unnecessary; math/rand documentation says
> The default Source is safe for concurrent use by multiple goroutines, but Sources created by NewSource are not.

where `math/rand.Read` and `math/rand.Seed` refer to the global source.

Reply:
So I think you're saying...we should not be setting a seed at all, and just rely on the default state?

Reply:
In this review comment I’m saying that the `randLock` variable doesn’t need to exist; it can just be dropped and the pre-existing non-locking can continue to be used without issues, making the rest of the PR smaller.

Reply:
This is the thing that got me thinking we needed locks:

https://pkg.go.dev/math/rand#Rand.Read

    "Read should not be called concurrently with any other Rand method."

Reply:
Yes, but that’s `rand.New().Read`, not `rand.Read`. We AFAICS use the latter one, https://pkg.go.dev/math/rand#Read .